content/docs/concepts/low-power.md

```diff
@@ -0,0 +1,69 @@
+
+The standard way to pause execution in Go is to use `time.Sleep()` function that is available in TinyGo too. Depending on the microcontroller architecture and effort put into it, the efficiency of this function can vary. The only thing certain: this is better than busy looping until a certain time has passed.  
+Still, with such high-level function as `time.Sleep()` it is not always possible to disable everything that consumes power as the decisions may depend on the application or the board used.
+
+Most microcontrollers have one or more "low power" modes, also known as "deep sleep" and, sometimes, "dormant".  
+In such modes microcontroller may consume as low as milli- or even micro-Watts.  
+TinyGo gives you low level access to hardware to try and make most of it.
+
+## Implementations
+
+Manufacturers implement different modes, depending of capabilites of their products and name these modes differently too.
+Hence no naming consistency across microcontrollers, unfortunately.
+
+Always consult "Power consumption" part of a respective datasheet to understand capabilities of your hardware.  
+When reading datasheets, keep in mind peripherals that draw power too. These include \[power\] LEDs, various sensors or co-processors usually present on development boards.
+
+Capabilities of different chips:
+- Some very good chips (like the nRF series) can go to a very low power state (just a few µA) with RAM retention and waking up from a low-power RTC.
+- Some limited chips (like the esp32) can't really "sleep" with low power -- they can disable the core and memory and reset after a predefined time losing state stored in memory.
+- Other chips need to use a special feature to go into deep sleep mode.
+
+Especially the distinction between waking up with or without RAM is important, because it greatly affects how you write software.
+
+The common way to wake up from deep sleep is to use [interrupts](compiler-internals/interrupts), such as timer, real-time-clock or an interrupt on a pin change.
+
+### RP2040
+
+Minimal achievable power consumption according to [RP2040 datasheet](https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf): 6mW to 12mW (0.19mA at 3.3V)
+
+### nRF Family
+
+[nRF series chips](https://www.nordicsemi.com/Products) are exceptionally efficient and is recommended choice for low-power projects.
+
+Minimal achievable power consumption is in the area of micro-Watts.
+
+Even with BLE broadcasting enabled it can be lower power than rp2040 in its lowest power state!
+
+The things needed to make nrf series chips efficient are:
+- Disable the UART with help of `-serial=none` flag;
+- Use the special sleep function of the SoftDevice;
+
+## The Way to Low Power
+
+Following steps shall help you achieve low power consumption:
+
+- Put a `for { arm.Asm("wfi") }` loop before anything else is initialized (at the beginning of [Reset_Handler](https://github.com/tinygo-org/tinygo/search?q=%22export+Reset_Handler%22)).  
```

Comment:
That loop is infinite? Should there be a `isInitialized()` in the for loop clause?

Reply:
As I understood @aykevl idea, this loop shall permanently lock the core in some specific place. And nothing need to be initialised, the asm call is fine as it is.
Then you move this lock down the code to find the place system starts eating more, and address that.

Reply:
Yes, it's just for testing. It's a way to make sure you start from a place where nothing is initialized.

Oh, now I remember something else: when a debugger is attached, that will usually block the low power state. So the best way to measure is to flash the code, then do a power cycle, and then measure. Probably worth adding to the documentation.

Reply:
Added about the debugger, thanks.